src/plugins/gs-flatpak.c

```diff
@@ -2393,6 +2393,19 @@ gs_flatpak_app_install (GsFlatpak *self,
 	return TRUE;
 }
 
+static gboolean
+error_is_ostree_missing_file_obj (GError *error)
+{
+	if (!g_error_matches (error, G_IO_ERROR,
+			      G_IO_ERROR_NOT_FOUND))
+		return FALSE;
+
+	/* XXX: matching the error messages is not ideal but we need to further
+	 * identify the error we are interested in */
+	return (g_strstr_len (error->message, -1, "No such file object") != NULL) ||
+		(g_strstr_len (error->message, -1, "Couldn't find file object") != NULL);
```

Comment:
Ugh, I understand why you're doing this (flatpak could return the same error code) but it's really ugly! I am amendable to accepting this for 3.1, but we should have a better plan going forward. Ideally returning a different error from ostree when this is the case.
Can you talk with upstream and open an issue about it?